workflow/rules/long_read.smk

```diff
@@ -32,82 +32,60 @@ rule count_fastq_reads:
         "echo $(( $(cat {input} | wc -l ) / 4)) > {output} 2> {log}"
 
 
-# Intermediate number of threads (4-8) achieve best speedup of a+btrimming.
-# For large files 8 threads help accelerate some, small files are processed faster with 4 threads.
-rule porechop_adapter_barcode_trimming:
+rule nanofilt:
     input:
         get_fastqs,
     output:
-        temp("results/{date}/trimmed/porechop/adapter_barcode_trimming/{sample}.fastq"),
-    conda:
-        "../envs/porechop.yaml"
+        temp("results/{date}/filtered/nanofilt/{sample}.fastq"),
     log:
-        "logs/{date}/trimmed/porechop/adapter_barcode_trimming/{sample}.log",
-    threads: 8
-    shell:
-        "porechop -i {input} -o {output} -t {threads} -v 1 > {log} 2>&1"
-
-
-rule customize_primer_porechop:
-    input:
-        get_artic_primer,
-    output:
-        temp("results/.indicators/replacement_notice.txt"),
+        "logs/{date}/nanofilt/{sample}.log",
+    params:
+        min_length=config["quality-criteria"]["ont"]["min-length-reads"],
+        min_PHRED=config["quality-criteria"]["ont"]["min-PHRED"],
     conda:
-        "../envs/primechop.yaml"
-    log:
-        "logs/customize_primer_porechop.log",
+        "../envs/nanofilt.yaml"
     shell:
-        "(cp {input} $CONDA_PREFIX/lib/python3.9/site-packages/porechop/adapters.py && "
-        'echo "replaced adpaters in adapter.py file in $CONDA_PREFIX/lib/python3.9/site-packages/porechop/adapters.py with ARTICv3 primers" > {output}) '
-        "2> {log}"
+        "NanoFilt --length {params.min_length} --quality {params.min_PHRED} --maxlength 700 {input} > {output} 2> {log}"
 
 
-# Using a low number of threads (2-4) speed up primer-trimming significantly (>2x), even for large files,
-# presumably due to the much higher number of target-sequences for trimming as compared
-# to barcode+adapter-trimming. However, using only one thread is again very slow.
-rule porechop_primer_trimming:
+rule downsample_and_trim_raw:
     input:
-        fastq_in=(
-            "results/{date}/trimmed/porechop/adapter_barcode_trimming/{sample}.fastq"
-        ),
-        repl_flag="results/.indicators/replacement_notice.txt",
+        primer=get_artic_primer,
+        reads="results/{date}/filtered/nanofilt/{sample}.fastq",
+        ref_genome="resources/genomes/main.fasta",
     output:
-        temp("results/{date}/trimmed/porechop/primer_clipped/{sample}.fastq"),
-    conda:
-        "../envs/primechop.yaml"
+        "results/{date}/norm_trim_raw_reads/{sample}/{sample}.cap.fasta",
+        "results/{date}/norm_trim_raw_reads/{sample}/{sample}.cap.clip.fasta",
```

Comment:
```suggestion
        temp("results/{date}/norm_trim_raw_reads/{sample}/{sample}.cap.fasta)",
        temp("results/{date}/norm_trim_raw_reads/{sample}/{sample}.cap.clip.fasta)",
```

Honestly, I have no idea about the runtime of notramp. If it doesn't take too long, I would vote for temping the output of it, bc of disk space.

Reply:
Morning. Notramp with capping and trimming takes about 15s on my linux laptop and about a minute on my mac (maybe faster after a restart). THh files will be around 15MB per sample (cause of the downsampling). You think that could become critical?